-Wl,-no-undefined is broken with openbsd's compiler for libc functions
(probably because we specifically avoid registering libc in NEEDED)

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -52,7 +52,7 @@ dnl FLTK library versions, currently 1.3.9
 FLTK_VERSION_MAJOR=1
 FLTK_VERSION_MINOR=3
 FLTK_VERSION_PATCH=9
-FL_DSO_VERSION=${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}
+dnl FL_DSO_VERSION=${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}
 FL_ABI_VERSION=${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}.0
 FLTK_VERSION=${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}.${FLTK_VERSION_PATCH}
 
@@ -968,7 +968,7 @@ case $host_os_gui in
 	    AC_MSG_WARN([Ignoring libraries "$X_PRE_LIBS" requested by configure.])
 	fi
 
-	LIBS="$LIBS -lX11 $X_EXTRA_LIBS"
+	LIBS="$LIBS -lX11 -lXdmcp -lXau $X_EXTRA_LIBS"
 	CFLAGS="$CFLAGS $X_CFLAGS"
 	CXXFLAGS="$CXXFLAGS $X_CFLAGS"
 	LDFLAGS="$X_LIBS $LDFLAGS"
@@ -1293,7 +1293,7 @@ if test -n "$GCC"; then
 	if test "x$with_optim" != x; then
 	    OPTIM="$with_optim $OPTIM"
 	else
-	    OPTIM="-Os $OPTIM"
+	    : # OPTIM="-Os $OPTIM"
 	fi
     fi
 
@@ -1323,18 +1323,6 @@ if test -n "$GCC"; then
 	 AC_MSG_RESULT(yes)],
 	[AC_MSG_RESULT(no)])
     CFLAGS="$OLDCFLAGS"
-
-dnl Make sure that shared libraries don't have undefined references
-    # See if ld supports -no-undefined...
-    AC_MSG_CHECKING([if ld supports -no-undefined])
-    OLDLDFLAGS="$LDFLAGS"
-    LDFLAGS="$LDFLAGS -Wl,-no-undefined"
-    AC_LINK_IFELSE(
-	[AC_LANG_PROGRAM([[]], [[]])],
-	[DSOFLAGS="$DSOFLAGS -Wl,-no-undefined"
-	 AC_MSG_RESULT(yes)],
-	[AC_MSG_RESULT(no)])
-    LDFLAGS="$OLDLDFLAGS"
 
     # See if ld supports -Bsymbolic-functions...
     AC_MSG_CHECKING([if ld supports -Bsymbolic-functions])
